### ISSUE TITLE
Error shown when adding device type without platform permission

### ISSUE THREAD

**Issue by llamafilm** [netbox-community/netbox#15271]
### Deployment Type

Self-hosted

### NetBox Version

v3.7.0

### Python Version

3.10

### Steps to Reproduce

1. Create a user with permissions limited to `dcim.device_type` and `dcim.manufacturer`.
2. Login as that user and create a new device type
3. Select any existing Manufacturer in the dropdown list

### Expected Behavior

I should be able to create a Device Type without error

### Observed Behavior

The UI shows this error:
<img width="1231" alt="Screenshot 2024-02-25 at 7 36 54 PM" src="https://github.com/netbox-community/netbox/assets/3991046/eee1950b-bfa9-4828-b273-f5479bb5961f">

**Comment by llamafilm** [comment]
Obviously I could work around this by adding `dcim.platform` permission to this user group.  But that would cause Platforms to appear in the sidebar.  I'm trying to reduce clutter in the sidebar for items we don't use.

**Comment by llamafilm** [comment]
A similar issue happens when creating a new device without having permission for `dcim.rack`

**Comment by abhi1693** [comment]
This is expected. And, you should still be able to create a device. The default platform field is optional and the error indicates that it was unable to fetch the list rather than unable to create device type.

**Comment by llamafilm** [comment]
Yes I know it's optional, but this makes for a bad user experience.  I already got one user asking for help because they didn't realize they could just ignore the error.

I'm trying to use permissions to hide unused fields, to indicate to each user which fields they are supposed to work with.  I would prefer that any unpermitted fields should be hidden completely.  I'm open to other ideas too, but showing a big red error message is unacceptable.

**Comment by abhi1693** [comment]
You should raise this as a FR if you want new functionality. Note that no new UI work will be implemented for v3 but you can add this for future work like v4.1
